CMakeLists.txt

```diff
@@ -20,7 +20,28 @@ else()
     find_package(Threads)
 
     set(CMAKE_CXX_FLAGS
-      "${CMAKE_CXX_FLAGS} -g -std=c++11 -Wall -Wpedantic")
+      "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wpedantic")
+endif()
+
+if (${VARIANT} STREQUAL "coverage")
+    set(CMAKE_CXX_FLAGS
+      "${CMAKE_CXX_FLAGS} -fprofile-arcs -ftest-coverage")
+    set(CMAKE_BUILD_TYPE RELWITHDEBINFO)
+    set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -lgcov")
+elseif (${VARIANT} STREQUAL "asan")
+    set(CMAKE_CXX_FLAGS
+      "${CMAKE_CXX_FLAGS} -fsanitize=address -fno-omit-frame-pointer")
+    set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=address")
+    set(CMAKE_BUILD_TYPE RELWITHDEBINFO)
+elseif (${VARIANT} STREQUAL "usan")
+    set(CMAKE_CXX_FLAGS
+      "${CMAKE_CXX_FLAGS} -fsanitize=undefined -fno-omit-frame-pointer")
+    set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=undefined")
+    set(CMAKE_BUILD_TYPE RELWITHDEBINFO)
+elseif (${VARIANT} STREQUAL "debug")
+    set(CMAKE_BUILD_TYPE DEBUG)
+elseif (${VARIANT} STREQUAL "release")
+    set(CMAKE_BUILD_TYPE RELEASE)
 endif()
 
 message ("cxx Flags: " ${CMAKE_CXX_FLAGS})
```

Comment:
We might want to remove that message. It looks like debug code that was accidentally left in.